### PR TITLE
Remove metric for tracking of connected takers

### PR DESCRIPTION
**Pull request by thomaseizinger**

### CODE REVIEW

**Comment by thomaseizinger** [comment]
This should not be merged until our metrics are updated to use https://github.com/itchysats/itchysats/pull/1273.

File: daemon/src/maker_inc_connections.rs
Comment (luckysori) [inline daemon/src/maker_inc_connections.rs]: 🤔 This seems okay for now, but should we consider creating a little `order::Actor` which takes care of this state? It seems like the order doesn't belong in the "connection" actor.

**Review by luckysori — APPROVED** [review]

File: daemon/src/maker_inc_connections.rs
Comment (thomaseizinger) [reply daemon/src/maker_inc_connections.rs]: But then we have to send requests back and forth again which lead to a deadlock in the previous version. I am unsure!

Comment (luckysori) [reply daemon/src/maker_inc_connections.rs]: I get your point and there's no need to change this with this PR anyway. Some thoughts (from now on I will talk about offers, since I think that's the right term we agreed on):

If the maker's `offer::Actor` can only receive a new offer from the HTTP API and respond to a `GetOffer` request from the `maker_inc_connections::Actor`, is there potential for a deadlock?

**Review by bonomat — CHANGES_REQUESTED** [review]
Please don't merge yet. Removing this will break some of the metrics we are recording, e.g. how many takers have connected and how many have opened a position.

**Comment by thomaseizinger** [comment]
> Please don't merge yet. Removing this will break some of the metrics we are recording, e.g. how many takers have connected and how many have opened a position.

Yep! I think we will want to wait for #1331 actually before we remove this functionality.

File: daemon/src/maker_inc_connections.rs
Comment (thomaseizinger) [reply daemon/src/maker_inc_connections.rs]: I think there can be if there is a circular dependency in general.

Comment (luckysori) [reply daemon/src/maker_inc_connections.rs]: Do you see a request with a direct response (in the context of `xtra`, a `Message` with a `Result` other than `()`) as a circular dependency?

Comment (thomaseizinger) [reply daemon/src/maker_inc_connections.rs]: If both actors send messages to each other, that is a circular dependency.

Comment (thomaseizinger) [reply daemon/src/maker_inc_connections.rs]: Perhaps a dedicated `offer::Actor` would not create a circular dependency, I haven't fully thought it through :)

**Review by luckysori — APPROVED** [review]

**Review by bonomat — APPROVED** [review]
Note: this is a breaking API change and we need to adopt maker-metrics accordingly.

**Comment by thomaseizinger** [comment]
> Note: this is a breaking API change and we need to adopt maker-metrics accordingly.

Yes. I am already testing this metric on testnet and it is working as expected there.

**Comment by thomaseizinger** [comment]
> Note: this is a breaking API change and we need to adopt maker-metrics accordingly.

We can wait with merging this until after the next deployment and adopt the dashboard first!

**Comment by bonomat** [comment]
> > Note: this is a breaking API change and we need to adopt maker-metrics accordingly.
> 
> We can wait with merging this until after the next deployment and adopt the dashboard first!

To be clear: I was talking about the maker-metrics binary we built on top of the daemon which uses this HTTP API.

**Comment by thomaseizinger** [comment]
> > > Note: this is a breaking API change and we need to adopt maker-metrics accordingly.
> > 
> > 
> > We can wait with merging this until after the next deployment and adopt the dashboard first!
> 
> To be clear: I was talking about the maker-metrics binary we built on top of the daemon which uses this HTTP API.

Yes I know. We should adopt everything else first and only merge this once nothing else is using this API.

**Comment by thomaseizinger** [comment]
It would be nice to remove this but `maker-metrics` has not been updated and it is not in the current focus, hence closing this.

**Pull request closed without merging**
